src/vsr/replica.zig

```diff
@@ -1485,14 +1594,7 @@ pub fn ReplicaType(
 
             const op = message.header.op;
             const slot = self.journal.slot_for_op(op);
-            const checksum: ?u128 = switch (message.header.timestamp) {
-                0 => null,
-                1 => message.header.context,
-                else => unreachable,
-            };
-
-            // Only the primary may respond to `request_prepare` messages without a checksum.
-            assert(checksum != null or self.primary_index(self.view) == self.replica);
```

Comment:
`request_prepare` with no checksum used to be used during repair to advance the head op. Now we use RSV/SV for that purpose instead, which means `on_request_prepare` doesn't need to handle to "no checksum" case. (Which is good, since it was very risky + I found a couple bugs while ripping it out).